client.go

```diff
@@ -119,6 +138,43 @@ type ReattachConfig struct {
 	Pid  int
 }
 
+// SecureConfig is used to configure a client to verify the integrity of an
+// executable before running. It does this by verifying the checksum is
+// expected. Hash is used to specify the hashing method to use when checksumming
+// the file.  The configuration is verified by the client by calling the
+// SecureConfig.Check() function.
```

Comment:
I'd recommend adding more color to this comment to explain how to do this securely. Simply setting this won't buy you "security" as you know (since it depends on how this data came in and how the binary was installed and so on) but I think it'd be prudent to be explicit about this here, nicely. :)